File: modules/expo-gander-swiss-army/ios/Visibility/ExpoGanderVisibilityViewModule.swift
import ExpoModulesCore

public class ExpoGanderVisibilityViewModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoGanderVisibilityView")

    AsyncFunction("updateActiveViewAsync") {
      VisibilityViewManager.shared.updateActiveView()
    }

    View(VisibilityView.self) {
      Events([
        "onChangeStatus"
      ])

      Prop("enabled") { (view: VisibilityView, prop: Bool) in
        view.enabled = prop
      }
    }
  }
}

File: modules/expo-gander-swiss-army/android/src/main/java/expo/modules/ganderswissarmy/visibilityview/ExpoGanderVisibilityViewModule.kt
package expo.modules.ganderswissarmy.visibilityview

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoGanderVisibilityViewModule : Module() {
  override fun definition() =
    ModuleDefinition {
      Name("ExpoGanderVisibilityView")

      AsyncFunction("updateActiveViewAsync") {
        VisibilityViewManager.updateActiveView()
      }

      View(VisibilityView::class) {
        Events(arrayOf("onChangeStatus"))

        Prop("enabled") { view: VisibilityView, prop: Boolean ->
          view.isViewEnabled = prop
        }
      }
    }
}
